### PR TITLE
Fix issue 61: Use importlib to replace imp due to deprecation warning

### PR DESCRIPTION
**Pull request by Chouvic**
This PR is to use "importlib" to replace the [deprecated "imp“ module](https://docs.python.org/3/library/imp.html) as described in [issue 61](https://github.com/behave-restful/behave-restful/issues/61).

The fix is based on [this answer.](https://stackoverflow.com/a/37124336/11607969)

### CODE REVIEW

**Comment by CurroRodriguez** [comment]
@Chouvic The feature tests are failing for your PR. It looks like the problem is caused by the change to use importlib that doesn't load the definitions correctly. I suggest that you run `bolt ft` in the repo, so you can see the feature tests running and the errors:

```
0 features passed, 0 failed, 0 skipped, 20 untested
0 scenarios passed, 0 failed, 0 skipped, 71 untested
0 steps passed, 0 failed, 0 skipped, 0 undefined, 221 untested
Took 0m0.000s
Traceback (most recent call last):
  File "/home/travis/virtualenv/python3.5.6/bin/bolt", line 11, in <module>
    sys.exit(run())
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/bolt/__init__.py", line 38, in run
    app.run()
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/bolt/_btapp.py", line 59, in run
    self.run_task(self.options.task)
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/bolt/_btapp.py", line 70, in run_task
    runner.run(task_name)
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/bolt/_btrunner.py", line 22, in run
    self._try_run_task(task)
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/bolt/_btrunner.py", line 51, in _try_run_task
    self._run_task(task)
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/bolt/_btrunner.py", line 63, in _run_task
    result = operation(config=config, context=self._context)
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/bolt/api.py", line 15, in __call__
    self._execute()
  File "/home/travis/build/behave-restful/behave-restful/behave_restful/bolt_behave_restful.py", line 39, in _execute
    if result != 0: raise bolt_api.TaskFailedError()
```

File: behave_restful/_utils.py [hunk 1]
```diff
@@ -28,11 +28,8 @@ def load_module(module_name):
...
     """
-    module_file = None
-    try:
-        module_file, pathname, description = imp.find_module(module_name)
-        module = imp.load_module(module_name, module_file, pathname, description)
-        return module
-    finally:
-        if module_file:
-            module_file.close()
+    module_spec = importlib.util.find_spec(module_name)
```
Comment (CurroRodriguez) [inline behave_restful/_utils.py:31]: I believe the problem is here. You're trying to find the spec and then use it to load the module. You just have to call `importlib.load_module(module_name)` and return the module. You don't need to raise any exception since it will raise the exception itself.

File: tests/test_utils.py [hunk 2]
```diff
@@ -28,6 +28,20 @@ def test_does_not_add_path_if_it_does_not_exist(self):
         assert_that(self.not_existing_path in sys.path).is_false()
 
 
+class TestLoadModule(unittest.TestCase):
+    def setUp(self):
+        self.test_module_name = "os"
```
Comment (CurroRodriguez) [inline tests/test_utils.py:33]: This kind of gives away the problem. You are expecting a module name where module_name is really a `.py` file name.

File: behave_restful/_utils.py (hunk 1 above)
Comment (Chouvic) [reply behave_restful/_utils.py]: Thanks, changed to [importlib.import_module](https://docs.python.org/3/library/importlib.html#importlib.import_module)(module_name).

File: tests/test_utils.py (hunk 2 above)
Comment (Chouvic) [reply tests/test_utils.py:33]: Do you have any idea about a better test_module_name?

**Comment by Chouvic** [comment]
Thanks for your review, I used `bolt ft` and it's working fine now.
Also, I encounter the following issue:
```(behave-restful) jun@jun-dell-7530:~/Desktop/Projects/behave-restful$ make -C docs html
make: Entering directory '/home/jun/Desktop/Projects/behave-restful/docs'
Running Sphinx v3.2.0

Exception occurred:
  File "/home/jun/.local/share/virtualenvs/behave-restful-nFZmzslZ/lib/python3.8/site-packages/sphinx/application.py", line 1084, in add_source_parser
    self.registry.add_source_parser(*args, **kwargs)
TypeError: add_source_parser() takes 2 positional arguments but 3 were given
The full traceback has been saved in /tmp/sphinx-err-ksdo2j7m.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
make: *** [Makefile:20: html] Error 2
make: Leaving directory '/home/jun/Desktop/Projects/behave-restful/docs'
```

I guess it is a version issue because of sphinx, thus I locked its version based on [your last successful build](https://travis-ci.org/github/behave-restful/behave-restful/jobs/661737209).

File: tests/test_utils.py (hunk 2 above)
Comment (CurroRodriguez) [inline tests/test_utils.py:33]: @Chouvic One of the reasons I didn't write unit tests for this is because writting unit tests that actually load a file are kind of weak. You may endup having false negatives if something has loaded the module earlier.

I think you should remove these tests. The code is tested through the feature tests because they load definition files, and if they can't be loaded, as you saw, the tests fail.

File: requirements.txt [hunk 3]
```diff
@@ -5,5 +5,5 @@ coverage>=4.1
 m2r>=0.2.1
 nose>=1.3.7
 pylint>=1.7.4
-sphinx>=1.3
-sphinx-rtd-theme>=0.1.9
\ No newline at end of file
+sphinx==2.4.4
```
Comment (CurroRodriguez) [inline requirements.txt:8]: Why did you need to update this dependency?

Comment (CurroRodriguez) [reply requirements.txt:8]: Nevermind, I found the problem myself.

File: tests/test_utils.py (hunk 2 above)
Comment (Chouvic) [reply tests/test_utils.py]: Yeah, that's correct. Tests removed.

**Comment by CurroRodriguez** [comment]
@chouvic Thanks for your contribution on this. Great job!!!

**Comment by Chouvic** [comment]
> @Chouvic Thanks for your contribution on this. Great job!!!

Thanks for your quick release. Now I have upgraded your package and the following warning is gone for good.

`/usr/local/lib/python3.6/dist-packages/behave_restful/_utils.py:5: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses import imp`
